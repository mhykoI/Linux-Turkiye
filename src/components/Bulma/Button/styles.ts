import { Styles } from 'react-jss';

export default {
  noBorder: {
    border: 'none',
  },
} as Styles;
