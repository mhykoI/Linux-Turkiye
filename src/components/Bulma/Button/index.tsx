import React from 'react';
import clsx from 'clsx';
import { createUseStyles } from 'react-jss';
import { Bulma } from '../../../models';
import styles from './styles';

const useStyles = createUseStyles(styles);

export interface ButtonProps {
  element: 'button' | 'a';
  anchor?: {
    href?: string;
    target?: '_blank' | '_self';
  };
  size?: Bulma.Size;
  color?: Bulma.Color;
  title?: string;
  noBorder?: boolean;
  outlined?: boolean;
  inverted?: boolean;
  rounded?: boolean;
  handlers?: { [handler: string]: (e?: any) => void };
}

const Button: React.FC<ButtonProps> = ({
  element: Element,
  anchor,
  title,
  color,
  noBorder,
  handlers,
  children,
  ...rest
}) => {
  const classes = useStyles();

  const anchorProps = Element === 'a' ? { ...anchor, 'aria-label': title } : {};
  const styleRest = Object.keys(rest).map((key) => ({ [`is-${key}`]: !!key }));

  return (
    <Element
      {...anchorProps}
      {...handlers}
      className={clsx(
        'button',
        { [classes.noBorder]: noBorder },
        { [`is-${color}`]: !!color },
        ...styleRest,
      )}
      title={title}
    >
      {children}
    </Element>
  );
};

export default Button;
