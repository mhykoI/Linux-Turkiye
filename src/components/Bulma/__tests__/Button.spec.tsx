import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../Button';

describe('<Button />', () => {
  test('should render correctly', () => {
    const tree = renderer
      .create(
        <Button
          element="a"
          anchor={{ href: 'https://linux.org' }}
          title="Deepin TÃ¼rkiye"
          rounded
          size="medium"
          color="success"
        >
          linux.org
        </Button>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
