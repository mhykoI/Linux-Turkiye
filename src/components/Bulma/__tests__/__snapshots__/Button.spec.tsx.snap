// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render correctly 1`] = `
<a
  aria-label="Deepin Türkiye"
  className="button is-success is-rounded is-size"
  href="https://linux.org"
  title="Deepin Türkiye"
>
  linux.org
</a>
`;
